pkg/controllers/workgenerator/controller.go

```diff
@@ -1245,11 +1245,20 @@ func extractResFromConfigMap(uConfigMap *unstructured.Unstructured) ([]fleetv1be
 		return nil, err
 	}
 	// the list order is not stable as the map traverse is random
-	for _, value := range configMap.Data {
+	for key, value := range configMap.Data {
+		// so we need to check the GVK and annotation of the selected resource
 		content, jsonErr := yaml.ToJSON([]byte(value))
 		if jsonErr != nil {
 			return nil, jsonErr
 		}
+		var uManifest unstructured.Unstructured
+		if unMarshallErr := uManifest.UnmarshalJSON(content); unMarshallErr != nil {
+			klog.ErrorS(unMarshallErr, "manifest has invalid content", "manifestKey", key, "envelopResource", klog.KObj(uConfigMap))
+			return nil, fmt.Errorf("the object with manifest key `%s` in evenlop config `%s` is malformatted, err: %w", key, klog.KObj(uConfigMap), unMarshallErr)
+		}
```

Comment:
Hi Ryan! Just a nit here: would it be good if we add a comment here mentioning that namespaced objects that do not have a namespace value specified would pass the check here but fail at the work applier end?

Reply:
So they will not have assigned the same namespace as the configMap?

Reply:
we won't assign the namespace since we don't know if it's a namespace scoped or cluster scoped object